utilities/utilities/api/base_classes/todoist.py

```diff
@@ -160,14 +157,33 @@ def delete_all_items_in_project(
                         continue
             if only_with_label and only_with_label not in task.labels:
                 continue
+            tasks_to_be_deleted.append(task)
 
-            self._delete_task(task_id=task.id)
-            tasks_deleted += 1
-
+        # checking if deletion desired & performing it
+        number_tasks_to_be_deleted = len(tasks_to_be_deleted)
         FILE_LOGGER.info(
-            "[todoist delete]", action=f"Deleted {tasks_deleted} tasks!"
+            "[todoist delete]",
+            action="Delete items in project",
+            project=project,
+            number_tasks_to_be_deleted=number_tasks_to_be_deleted,
         )
-        return tasks_deleted
+        if number_tasks_to_be_deleted > 0:
+            choice = YesNoChoices.ask_yes_no(
+                f"... delete {number_tasks_to_be_deleted} tasks?"
+            )
+            if choice == YesNoChoices.no:
+                FILE_LOGGER.warning(
+                    "[todoist delete]",
+                    action="You did NOT delete tasks",
+                    number_tasks_to_be_deleted=number_tasks_to_be_deleted,
+                )
+                return 0
```

Comment:
@AkagiShigeru we probably want to implement something else here if we choose not to delete tasks. Like either here raise an error so that it doesn't continue to process things or in the relevant grocery/menu code. What would you prefer?

Reply:
- [ ] whatever we choose to do should be covered by a new test.